src/zenml/zen_server/routers/plugin_endpoints.py

```diff
@@ -37,93 +44,61 @@
-    response_model=List[str],
+    response_model=Page[BasePluginFlavorResponse],
     responses={401: error_response, 404: error_response, 422: error_response},
 )
 @handle_exceptions
-def list_plugin_types(
-    flavor_name: str,
+def list_flavors(
+    type: PluginType,
+    subtype: PluginSubType,
+    page: int = PAGINATION_STARTING_PAGE,
+    size: int = PAGE_SIZE_DEFAULT,
+    hydrate: bool = False,
     _: AuthContext = Security(authorize),
-) -> List[str]:
+) -> Page[BasePluginFlavorResponse]:
     """Returns all event flavors.
 
-    Returns:
-        All flavors.
-    """
-    return plugin_flavor_registry.get_available_types_for_flavor(
-        flavor_name=flavor_name
-    )
-
-
-# -------------------- Plugin SubTypes --------------------
-
-
-@plugin_router.get(
-    "/{flavor_name}/{plugin_type}",
-    response_model=List[str],
-    responses={401: error_response, 404: error_response, 422: error_response},
-)
-@handle_exceptions
-def list_plugin_subtypes(
-    flavor_name: str,
-    plugin_type: PluginType,
-    _: AuthContext = Security(authorize),
-) -> List[str]:
-    """Returns all subtypes.
+    Args:
+        type: The type of Plugin
+        subtype: The subtype of the plugin
+        page: Page for pagination (offset +1)
+        size: Page size for pagination
+        hydrate: Whether to hydrate the response bodies
 
     Returns:
-        All subtypes for the given flavor and type.
+        A page of flavors.
     """
-    return plugin_flavor_registry.get_available_subtypes_for_flavor_and_type(
-        flavor=flavor_name, _type=plugin_type
+    flavors = plugin_flavor_registry.list_available_flavor_responses_for_type_and_subtype(
+        _type=type, subtype=subtype, page=page, size=size, hydrate=hydrate
     )
+    return flavors
 
 
 # -------------------- Flavors --------------------
 
 
 @plugin_router.get(
-    "/{flavor_name}/{plugin_type}/{plugin_subtype}",
+    "/{name}",
     responses={401: error_response, 404: error_response, 422: error_response},
 )
 @handle_exceptions
 def get_flavor(
-    flavor_name: str,
-    plugin_type: PluginType,
-    plugin_subtype: PluginSubType,
+    name: str,
+    type: PluginType = Query(..., alias="type"),
+    subtype: PluginSubType = Query(..., alias="subtype"),
     _: AuthContext = Security(authorize),
-):
+) -> BasePluginFlavorResponse:
     """Returns the requested flavor.
 
     Args:
-        flavor_name: Name of the flavor.
-        plugin_type: Type of Plugin
-        plugin_subtype: Subtype of Plugin
+        name: Name of the flavor.
+        type: Type of Plugin
+        subtype: Subtype of Plugin
 
     Returns:
         The requested flavor response.
     """
-    # TODO: Figure out if typing of the response can be reintroduced
-    return plugin_flavor_registry.get_flavor_class(
-        flavor=flavor_name, _type=plugin_type, subtype=plugin_subtype
-    ).get_plugin_flavor_response_model()
+    plugin_flavor = plugin_flavor_registry.get_flavor_class(
```

Comment:
These endpoints do not have any RBAC checks at the moment, but I guess there isn't really anything worth hiding in the flavors anyway?

Reply:
Yeah I think the flavors themselves need no RBAC - you're allows to know a plugin exists and what is needed to configure one